internal/handler/msg_getmore.go

```diff
@@ -266,3 +273,81 @@ func (h *Handler) makeNextBatch(c *cursor.Cursor, batchSize int64) (*types.Array
 
 	return nextBatch, nil
 }
+
+// awaitData stops the goroutine, and waits for a new data for the cursor.
+// If there's a new document, or the maxTimeMS have passed it returns the nextBatch.
+func (h *Handler) awaitData(ctx context.Context, c *cursor.Cursor, maxTimeMS, batchSize int64) (resBatch *types.Array, err error) {
+	data := c.Data.(*findCursorData)
+
+	closer := iterator.NewMultiCloser()
+
+	startTime := time.Now()
+	sleepDur := time.Duration(maxTimeMS) * time.Millisecond
+
+	ctx, cancel := context.WithTimeout(ctx, sleepDur)
+	defer cancel()
+
+	defer func() {
+		if err == nil {
+			return
+		}
+
+		if errors.Is(err, context.DeadlineExceeded) {
+			err = nil
+			return
+		}
+
+		err = lazyerrors.Error(err)
+		resBatch = nil
+		return
+	}()
+
+	for {
+		c.Close()
```

Comment:
How does the whole code work overall?

Let's say, there is a collection X with some data. I query this data with `awaitData = true`.
I got some data, but because `awaitData` is set as `true`, I just continue waiting for the data. As soon as there is new data, I should receive it in addition to what I already received, only the new data, nothing else. Then I received that new data and I continue waiting. When there is some new data I should receive that new data again. What I don't understand in this code is how you ensure that next time I call `getMore` with `awaitData`, I only receive the new data? I expected some `recordId` or something like this being involved. Or do I misunderstand something?

Reply:
(Also I see you call `c.Close()` but don't I need this cursor to actually get more data?..)

Reply:
Yeah, I think my understanding is not quite right.

Reply:
> (Also I see you call c.Close() but don't I need this cursor to actually get more data?..)

`c.Close()` if tailable, doesn't close whole cursor, but only underlying iterator.
We change the iterator on every for loop iteration, as we do new backend queries.

The overusage of `Close` is a result of bugs from previous week, as in some cases underlying iterator wasn't closed properly which resulted in panic. I'm currently cleaning up everything, so some `Close` calls will be removed, however its overusage shouldn't result in any issues so I think it's ready to run your tests